frontend/src/reducer/searchResult.js

```diff
@@ -17,18 +18,22 @@ const defaultResult = {
 }
 
 export default function searchResult (state = defaultResult, action) {
+  function uniqueResults (allResults) {
+    return uniqBy(allResults, 'content_id')
+  }
+
   switch (action.type) {
     case `${SET}/${SEARCH_RESULTS_LIST}`:
-      return {...state, resultsList: action.newSearchResultsList}
+      return {...state, resultsList: uniqueResults(action.newSearchResultsList)}
 
     case `${APPEND}/${SEARCH_RESULTS_LIST}`:
-      return {...state, resultsList: [...state.resultsList, ...action.appendSearchResultsList]}
+      return {...state, resultsList: uniqueResults([...state.resultsList, ...action.appendSearchResultsList])}
 
     case `${SET}/${SEARCHED_KEYWORDS}`:
       return {...state, searchedKeywords: action.searchedKeywords}
 
     case `${SET}/${CURRENT_NUMBER_SEARCH_RESULTS}`:
-      return {...state, currentNumberSearchResults: action.currentNumberSearchResults}
+      return {...state, currentNumberSearchResults: state.resultsList.length}
```

Comment:
since this reducer doesn't use the parameter action anymore, it should be removed from the action in action-creators.sync.js too.
But if it is now only calculated from resultsList.length, it's better to use a constant equals to props.resultsList.length in the component where you need it